packages/core/src/provider/base.ts

```diff
@@ -589,6 +601,17 @@ export class SmartAccountProvider<
     return this.account !== undefined;
   };
 
+  /*
+   * Note that the connected account's entryPointAddress always takes the precedence
+   */
+  getEntryPointAddress = (): Address => {
+    return (
+      this.account?.getEntryPointAddress() ??
+      this.entryPointAddress ??
+      getDefaultEntryPointAddress(this.chain)
+    );
+  };
```

Comment:
i think it should be this order so we prioritize the current object's entryPoint over its account. it shouldn't matter though if they're the same, but just in case.

```suggestion
  getEntryPointAddress = (): Address => {
    return (
      this.entryPointAddress ??
      this.account?.getEntryPointAddress() ??
      getDefaultEntryPointAddress(this.chain)
    );
  };
```